packages/rOpenScPCA/R/calculate-clusters.R

```diff
@@ -0,0 +1,106 @@
+#' Calculate graph-based clusters from a provided matrix
+#'
+#' This function is provided to simplify application of bluster package clustering functions on OpenScPCA data.
+#' In particular, this function runs bluster::clusterRows() with the bluster::NNGraphParam() function.
+#' Note that defaults for some arguments may differ from the bluster::NNGraphParam() defaults.
+#' Specifically, the clustering algorithm defaults to "louvain" and the distance measure to "jaccard"
+#' to align with common practice in scRNA-seq anaylsis.
+#'
+#' @param mat Matrix, usually of PCs, where each row is a cell. Matrix must have rownames of cell ids (e.g., barcodes)
+#' @param algorithm Clustering algorithm to use. Must be one of "louvain" (default), "walktrap", or "leiden".
+#' @param weighting Weighting scheme to use. Must be one of "jaccard" (default), "rank", or "number"
+#' @param nn Number of nearest neighbors. Default is 10.
+#' @param resolution Resolution parameter used by louvain and leiden clustering only. Default is 1.
+#' @param objective_function Leiden-specific parameter for whether to use the Constant Potts Model ("CPM"; default) or "modularity"
+#' @param seed Random seed to set for clustering. Default is 2024.
+#' @param cluster_args List of additional arguments to pass to the chosen clustering function. Only single-length values will be used.
```

Comment:
Slight reword.
```suggestion
#' @param cluster_args List of additional arguments to pass to the chosen clustering function. 
#'   Only single values for each argument are supported (no vectors or lists).
```

Reply:
well, "everything" is a vector, but i get it.